targets/nrf5x/bluetooth.c

```diff
@@ -391,12 +391,30 @@ static void on_ble_evt(ble_evt_t * p_ble_evt)
 
         break;
 
-      case BLE_GAP_EVT_RSSI_CHANGED: {
-        JsVar *evt = jsvNewFromInteger(p_ble_evt->evt.gap_evt.params.rssi_changed.rssi);
-        if (evt) jsiQueueObjectCallbacks(execInfo.root, BLE_RSSI_EVENT, &evt, 1);
-        jsvUnLock(evt);
-        jshHadEvent();
-      } break;
+      case BLE_GAP_EVT_RSSI_CHANGED: 
+#if CENTRAL_LINK_COUNT>0
+        if (m_central_conn_handle == p_ble_evt->evt.gap_evt.conn_handle) {
+          JsVar *gattServer = bleGetActiveBluetoothGattServer();
+          if (gattServer) {
+            JsVar *rssi = jsvNewFromInteger(p_ble_evt->evt.gap_evt.params.rssi_changed.rssi);
+            JsVar *bluetoothDevice = jsvObjectGetChild(gattServer, "device", 0);
+            if (bluetoothDevice) {
+              jsvObjectSetChild(bluetoothDevice, "rssi", rssi);
+            }
+            jsiQueueObjectCallbacks(gattServer, BLE_RSSI_EVENT, &rssi, 1);
+            jshHadEvent();
+            jsvUnLock(rssi);
```

Comment:
Super minor, but you know you can do UnLock3?

At one point I had a bit of a blitz - you'd be amazed how much Flash memory this frees up :)